File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        int count=1;
        for(int i=0;i<n-1;i++){
            if(nums[i]==nums[i+1]){
                count++;
            }
            else{
                count=1;
            }
            if(count>n/2) return nums[i];
        }
        return nums[n-1];
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer>map=new HashMap<>();
        int n=nums.length;
        int maximum=Integer.MIN_VALUE;
        int ans=-1;
        for(int i=0;i<n;i++){
            int num=nums[i];
            map.put(num,map.getOrDefault(num,0)+1);
            if(map.get(num)>maximum){
                maximum=map.get(num);
                ans=nums[i];
            }
        }
        return ans;
    }
}